docs/translating.md

```diff
@@ -76,25 +76,104 @@ To ensure we always internationalize our JSON errors messages, the
 Zulip linter (`tools/lint-all`) checks for correct usage.
 
 ## Frontend Translations
-The first step in translating the frontend is to create the translation
-files using `python manage makemessages`. This command will create
-translation files under `static/locale`, the location can be changed by
-passing an argument to the command, however make sure that the location is
-publicly accessible since these files are loaded through XHR in the
-frontend which will only work with publicly accessible resources.
+Zulip uses the [i18next][] library for the frontend translations. To make
+the frontend translations work, the first step is to create the JSON
+formatted [resource][] files using the `python manage makemessages`
+command. This command will create resource files under `static/locale`.
+The location can be changed by passing an argument to the command (see
+the help for the command for further details). However, make sure that
+the location is publicly accessible since these files are loaded through
+XHR in the frontend which will only work with publicly accessible resources.
+
+The second step is to upload the resource files to Transifex using
+the `tx push -s -a` command.
+
+The final step is to get the updated resource files from Transifex using
+the `tx pull -a` command. This command will download the resource files
+from Transifex and replace your local resource files with them.
+
+### Marking Strings for Translation
```

Comment:
>Since these files should be tracked in git, perhaps we should instruct people in that. If I'm a Zulip developer, and I've made user-visible changes in Zulip, I should make sure to run python manage.py makemessages and git add those files before I make a pull request, right?

I think the person who is responsible for uploading and downloading the resources from the Transifex should update the Git repository once new translations are available. Committing the `translations.json` file in a new PR won't serve any immediate purpose.